pkgs/development/python-modules/dm-control/default.nix

```
@@ -0,0 +1,99 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, absl-py
+, mujoco
+, pyparsing
+, pythonRelaxDepsHook
+, setuptools
+, wheel
+, dm-env
+, dm-tree
+, fsspec
+, glfw
+, h5py
+, lxml
+, mock
+, nose
+, nose-xunitmp
+, numpy
+, pillow
+, protobuf
+, pyopengl
+, requests
+, scipy
+, tqdm
+, etils
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "dm-control";
+  version = "1.0.18";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "google-deepmind";
+    repo = "dm_control";
+    rev = "refs/tags/${version}";
```

```suggestion
    rev = version;
```

https://github.com/NixOS/nixpkgs/blob/65d153261e9f006909d755d4c8e90058f34080c2/CONTRIBUTING.md?plain=1#L722-L736

---

Waiting for @DontEatOreo before proceeding further. I have no opinion on this. One or the other it's fine for me.

---

I think @GaetanLepage raises a valid concern, and actually, I agree with that committer. In that case, you can ignore this change. Also, could you please link to that committer's comment about this?

On another note, maybe there should be a PR that clarifies this in the manual or somewhere else? I'm currently looking at the [manual](https://nixos.org/manual/nixpkgs/unstable/ ) and there are examples of `rev = version;` in many places. [Searching for `"rev = version;"` on nixpkgs reveals it's widely used as well](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+%22rev+%3D+version%3B%22+path%3A%2F%5Epkgs%5C%2F%2F&type=code)

---

Fair enough, on my side I'll merge this PR once CI is done then.

---

> I think @GaetanLepage raises a valid concern, and actually, I agree with that committer. In that case, you can ignore this change. Also, could you please link to that committer's comment about this?

If I remember well, this was @SuperSandro2000, but I wouldn't be able to find the comment as it was ~1.5 years ago.

> On another note, maybe there should be a PR that clarifies this in the manual or somewhere else? I'm currently looking at the [manual](https://nixos.org/manual/nixpkgs/unstable/) and there are examples of `rev = version;` in many places. [Searching for `"rev = version;"` on nixpkgs reveals it's widely used as well](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+%22rev+%3D+version%3B%22+path%3A%2F%5Epkgs%5C%2F%2F&type=code)

I asked around and no-one seems to have a strong preference.
Relevant work: https://github.com/NixOS/nixpkgs/pull/153386